crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -894,6 +894,12 @@ impl<'interner> TypeChecker<'interner> {
                             Err(TypeCheckError::InvalidBitwiseOperationOnField { span })
                         } else if other.is_bindable() {
                             Err(TypeCheckError::AmbiguousBitWidth { span })
+                        } else if other.is_signed() {
+                            Err(TypeCheckError::TypeCannotBeUsed {
+                                typ: other,
+                                place: "bit shift",
+                                span,
```

Comment:
This will issue an error saying a signed type cannot be used in a bit shift even if the operation was not a bit shift. We need another check in this condition that the bitwise operator is a bitshift

Reply:
My bad, I got confused by the is_bitwise() guard. I have added an explicit check for bit_shift.